0.13.3 on conda and fresh git-annex: duplicated output picked up within test_runner_parametrized_protocol?

@notestaff working on adding downstream testing back to git-annex package on conda-forge which is great!

datalad testing is not yet green.
[Full log](https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/209014/logs/13) (160MB huge  due to annexremote testing verbosity, unrelated), but then testing of datalad fails with a single (better than if many) test failing

```
2020-09-14T06:57:15.1594851Z Versions: annexremote=1.4.3 appdirs=1.4.3 boto=2.49.0 cmd:7z=16.02 cmd:annex=8.20200908-ga853039 cmd:bundled-git=2.28.0 cmd:git=2.28.0 cmd:system-git=2.28.0 cmd:system-ssh=8.3p1 exifread=2.1.2 humanize=2.6.0 iso8601=0.1.13 keyring=21.4.0 keyrings.alt=3.1 msgpack=1.0.0 mutagen=1.41.1 patoolib=1.12 requests=2.24.0 tqdm=4.49.0 wrapt=1.12.1
2020-09-14T06:57:15.1601769Z Obscure filename: str=b' "\';a&b&c\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 `| ' repr=' "\';a&b&cΔЙקﻡ๗あ `| '
2020-09-14T06:57:15.1608345Z Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
2020-09-14T06:57:15.1616112Z Environment: PYTHONNOUSERSITE='1' PYTHONUNBUFFERED='1' PATH='$PREFIX:$PREFIX/bin:$PREFIX/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/conda/bin' LANG='en_US.UTF-8' GIT_BRANCH='merge' PYTHON='$PREFIX/bin/python' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch'"
2020-09-14T06:57:15.2906560Z Exception ignored in: <function _Database.close at 0x7fa3c4c100d0>
2020-09-14T06:57:15.2907546Z Traceback (most recent call last):
2020-09-14T06:57:15.2909376Z   File "/home/conda/feedstock_root/build_artifacts/git-annex_1600056447586/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/dbm/dumb.py", line 274, in close
2020-09-14T06:57:15.2910430Z     self._commit()
2020-09-14T06:57:15.2911738Z   File "/home/conda/feedstock_root/build_artifacts/git-annex_1600056447586/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/dbm/dumb.py", line 129, in _commit
2020-09-14T06:57:15.2913031Z     with self._io.open(self._dirfile, 'w', encoding="Latin-1") as f:
2020-09-14T06:57:15.2914219Z FileNotFoundError: [Errno 2] No such file or directory: '/tmp/datalad_temp_y1aehq1r/.config/datalad/cookies.dir'
2020-09-14T06:57:15.2914736Z
2020-09-14T06:57:15.2915093Z ======================================================================
2020-09-14T06:57:15.2915566Z FAIL: datalad.tests.test_witless_runner.test_runner_parametrized_protocol
2020-09-14T06:57:15.2916282Z ----------------------------------------------------------------------
2020-09-14T06:57:15.2916765Z Traceback (most recent call last):
2020-09-14T06:57:15.2918077Z   File "/home/conda/feedstock_root/build_artifacts/git-annex_1600056447586/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
2020-09-14T06:57:15.2919171Z     self.test(*self.arg)
2020-09-14T06:57:15.2920772Z   File "/home/conda/feedstock_root/build_artifacts/git-annex_1600056447586/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.8/site-packages/datalad/tests/test_witless_runner.py", line 149, in test_runner_parametrized_protocol
2020-09-14T06:57:15.2922014Z     eq_(res['stdout'], '5')
2020-09-14T06:57:15.2922684Z AssertionError: '55' != '5'
2020-09-14T06:57:15.2923417Z -------------------- >> begin captured logging << --------------------
2020-09-14T06:57:15.2924415Z asyncio: DEBUG: Using selector: EpollSelector
2020-09-14T06:57:15.2925346Z --------------------- >> end captured logging << ---------------------
2020-09-14T06:57:15.2925748Z
2020-09-14T06:57:15.2926393Z ----------------------------------------------------------------------
2020-09-14T06:57:15.2926856Z Ran 1163 tests in 1886.415s
2020-09-14T06:57:15.2927141Z
2020-09-14T06:57:15.2927419Z FAILED (SKIP=122, failures=1)
```

nope -- unfortunately it is rare but still not gone: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=489377&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=b2a8456a-fb11-5506-ca32-5ccd32538dc0 while building 0.16.1 against python 3.8
```
======================================================================
FAIL: datalad.runner.tests.test_witless_runner.test_runner_parametrized_protocol
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/datalad_1649524824072/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.8/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/conda/feedstock_root/build_artifacts/datalad_1649524824072/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.8/site-packages/datalad/runner/tests/test_witless_runner.py", line 202, in test_runner_parametrized_protocol
    eq_(res['stdout'], '5')
AssertionError: '55' != '5'
```

may be @christian-monch  has an idea?  (meanwhile I will force push rewrite to trigger a rerun of CI)

--- a/datalad/runner/tests/test_witless_runner.py
+++ b/datalad/runner/tests/test_witless_runner.py
@@ -194,18 +194,12 @@ def test_runner_parametrized_protocol():
             super().pipe_data_received(fd, self.value)
 
     res = runner.run(
-        py2cmd('print(1)'),
+        py2cmd('print(1, end="")'),
         protocol=ProtocolInt,
         # value passed to protocol constructor
         value=b'5',
     )
-    try:
-        eq_(res['stdout'], '5')
-    except AssertionError:
-        # TODO: remove when dropping support for python3.8
-        if res['stdout'] == '55' and sys.version_info[:2] == (3, 8):
-            pytest.xfail("got 55 and not 5, see https://github.com/datalad/datalad/issues/4921")
-        raise
+    eq_(res['stdout'], '5')
 
 
 @integration  # ~3 sec

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -2434,7 +2434,7 @@ def test_files_split(topdir=None, *, cls):
         os.unlink(f)
         with open(f, 'w') as f:
             f.write('1')
-    dl.save(dataset=r.path, path=dirs)
+    dl.save(dataset=r.path, path=dirs, result_renderer="disabled")
 
 
 @skip_if_on_windows

